tests/test_tokenization_common.py

```diff
@@ -1367,6 +1367,40 @@ def test_special_tokens_mask_input_pairs(self):
                 filtered_sequence = [x for x in filtered_sequence if x is not None]
                 self.assertEqual(encoded_sequence, filtered_sequence)
 
+    def test_truncation_side_in_kwargs(self):
+        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
+            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
+                truncation_side = "right"
+                kwargs.update({"truncation_side": truncation_side})
+
+                if self.test_rust_tokenizer:
+                    tokenizer_r = self.rust_tokenizer_class.from_pretrained(pretrained_name, **kwargs)
+                    self.assertEqual(tokenizer_r.truncation_side, truncation_side)
+
```

Comment:
Quick note on test readability.

Tests only asset is that they are readable, meaning you should know just by looking at them if they are correct or not.
That's why IMHO they should contain as little logic as possible (no `if` no `for`).

The iteration over ALL tokenizer classes is necessary to ensure some things are consistent across HF ecosystem.

But, my  point is that 

```python
 kwargs.update({"truncation_side": truncation_side})
```
is actually modifiying state much earlier that where the test actually lies, also 

```python
self.assertEqual(tokenizer_r.truncation_side, truncation_side)
```
Actually tests against a variable and not a value, meaning you can do a later update to the code, that actually is BREAKING and you're not going to see it, because the variable gets updated too by the code and the tests, will being unchanged has actually changed values, hiding it entirely the misbehavior from the test.


```suggestion
                if self.test_rust_tokenizer:
                    tokenizer_r = self.rust_tokenizer_class.from_pretrained(pretrained_name, truncation_side="left", **kwargs)
                    self.assertEqual(tokenizer_r.truncation_side, "left")
```

Is IMHO, a much better test. No state, no variable used as the reference value.

All the `ifs` everywhere are also littering readability IMO. Not sure what's the best course of action is there (what's in `self.tokenizers_list`, doesn't it also contains information about the supported classes ?)

Reply:
> Actually tests against a variable and not a value, meaning you can do a later update to the code, that actually is BREAKING and you're not going to see it, because the variable gets updated too by the code and the tests, will being unchanged has actually changed values, hiding it entirely the misbehavior from the test.

This is indeed a good point. I modified the test according to the template you propose. I had in mind that the arguments could possibly be already present in kwargs, but if it is the case we will override the test for the tokenizer in question.

> Tests only asset is that they are readable, meaning you should know just by looking at them if they are correct or not.
That's why IMHO they should contain as little logic as possible (no if no for).

> All the ifs everywhere are also littering readability IMO. Not sure what's the best course of action is there (what's in self.tokenizers_list, doesn't it also contains information about the supported classes ?)

Looking at your comments on readability, I understand them - I guess it's also a matter of preference. I've already reorganized the test to reduce the number of ifs to 2. Is that better? Unforthunatly, `self.tokenizers_list` only contain the fast class that is why we need to use `if self.test_rust_tokenizer` and `test_slow_tokenizer`. Regarding the `for` loop, I used it because it's the same template used in a lot of other tests of this class, So, I think that for this PR to keep the same format is adapted.

The other alternative I see for this test would consist in splitting it in 2:
```python
    def test_truncation_side_in_kwargs_slow(self):
        if not self.test_slow_tokenizer:
            continue

        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
                tokenizer_p = self.tokenizer_class.from_pretrained(
                    pretrained_name, truncation_side="left", **kwargs
                )
                self.assertEqual(tokenizer_p.truncation_side, "left")

                tokenizer_p = self.tokenizer_class.from_pretrained(
                    pretrained_name, truncation_side="right", **kwargs
                )
                self.assertEqual(tokenizer_p.truncation_side, "right")

                self.assertRaises(
                    ValueError,
                    self.tokenizer_class.from_pretrained,
                    pretrained_name,
                    truncation_side="unauthorized",
                    **kwargs,
                )

    def test_truncation_side_in_kwargs_fast(self):
        if not self.test_rust_tokenizer:
            continue

        for tokenizer, pretrained_name, kwargs in self.tokenizers_list:
            with self.subTest(f"{tokenizer.__class__.__name__} ({pretrained_name})"):
                tokenizer_r = self.rust_tokenizer_class.from_pretrained(
                    pretrained_name, truncation_side="left", **kwargs
                )
                self.assertEqual(tokenizer_r.truncation_side, "left")

                tokenizer_r = self.rust_tokenizer_class.from_pretrained(
                    pretrained_name, truncation_side="right", **kwargs
                )
                self.assertEqual(tokenizer_r.truncation_side, "right")

                self.assertRaises(
                    ValueError,
                    self.rust_tokenizer_class.from_pretrained,
                    pretrained_name,
                    truncation_side="unauthorized",
                    **kwargs,
                )
```

Reply:
I agree with your current choice. My personal preference is that it's the best compromise.

It's really a taste opinion between all 3 variants. I was stating that they hindered readability, but sometimes it's worth the price (for instance to keep consistency with other tests, or other considerations)

> Looking at your comments on readability, I understand them - I guess it's also a matter of preference. I've already reorganized the test to reduce the number of ifs to 2. Is that better? Unforthunatly, self.tokenizers_list only contain the fast class that is why we need to use if self.test_rust_tokenizer and test_slow_tokenizer. Regarding the for loop, I used it because it's the same template used in a lot of other tests of this class, So, I think that for this PR to keep the same format is adapted.

Thanks for reminding me about those variables, I wasn't sure without checking but it does make sense. And your point is clear. Fully agree here.